### PR TITLE
Add StringFormat tests and compat bugs

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

File: tests/teststringformat.c [hunk 1]
```diff
@@ -0,0 +1,742 @@
...
+	if (isnan (v1))
+		return isnan (v2);
+
+	if (isinf (v1))
+		return isinf (v2);
+
+	return fabs (v1 - v2) < 0.0001;
+}
+
+void verifyMatrix (GpMatrix *matrix, REAL e1, REAL e2, REAL e3, REAL e4, REAL e5, REAL e6)
```
Comment (akoeplinger) [inline tests/teststringformat.c]: use testhelper.h ?

File: tests/teststringformat.c [hunk 2]
```diff
@@ -0,0 +1,742 @@
...
+        fprintf (stderr, "Actual:   %d\n", actual);
+
+#if !defined(_WIN32)
+				abort ();
+#endif
+    }
+}
+
+#define assertEqualFloat(actual, expected) assertEqualFloatImpl (actual, expected, __FILE__, __LINE__)
+void assertEqualFloatImpl (REAL actual, REAL expected, const char *file, INT line)
```
Comment (akoeplinger) [inline tests/teststringformat.c]: same here.

File: src/stringformat-private.h [hunk 3]
```diff
@@ -44,6 +44,7 @@ struct _StringFormat {
 	StringFormatFlags	formatFlags;
 	StringTrimming		trimming;
 	StringDigitSubstitute	substitute;
+	LANGID		language;
```
Comment (akoeplinger) [inline src/stringformat-private.h]: please fix indendation so it lines up with the others

Comment (hughbe) [reply src/stringformat-private.h]: I am the worst at indentation, sorry! For some reason VS code sometimes gets it right but often doesn't. It also doesn't seem to be able to line up tabs correctly

Comment (akoeplinger) [reply src/stringformat-private.h]: yeah I see the same in VS Code. nevermind ❤️

**Pull request merged**